richtextfx-demos/src/main/java/org/fxmisc/richtext/demo/hyperlink/TextHyperlinkArea.java

```diff
@@ -0,0 +1,69 @@
+package org.fxmisc.richtext.demo.hyperlink;
+
+import javafx.geometry.VPos;
+import org.fxmisc.richtext.GenericStyledArea;
+import org.fxmisc.richtext.TextExt;
+import org.fxmisc.richtext.model.ReadOnlyStyledDocument;
+import org.fxmisc.richtext.model.StyledText;
+import org.fxmisc.richtext.model.TextOps;
+import org.reactfx.util.Either;
+
+import java.util.function.Consumer;
+
+public class TextHyperlinkArea extends GenericStyledArea<ParStyle, Either<StyledText<TextStyle>, Hyperlink<TextStyle>>, TextStyle> {
```

Comment:
I also reworked `StyledText`'s `equals` method to return true when both text objects are empty since `createText` will sometimes create an empty StyledText object but not use the one we want it to use (emptySeg).

Also, I'm not sure if our current pattern/approach is correct in one aspect: the empty segment's style. Technically, the segment is only empty in text, but not in style. The style is used when inserting text without specifying the style. The code will look at the style at the position of insertion and use that style for the text being inserted.

I think we need to update the code to use this approach:
````java
public Style getStyleForInsertion(int position) {
 if (useInitialTextstyle) {
  return initialTextStyle;
 } else {
  return getStyleAtPosition(position);
 }
}

public Style getStyleAtPosition(int position) {
 Optional<Style> s = getStyleAtPosition0(position);
 if (s.isPresent()) {
  return s.get();
 } else {
  return getStyleAtPosition(position - 1);
 }
}

public Optional<Style> getStyleAtPosition0(int position) {
 Style style = content.getStyleAtPosition(position);
 return Optional.ofNullable(style);
}
````